newIDE/app/src/MainFrame/Preferences/LanguageDialog.js

```diff
@@ -1,163 +1,77 @@
 // @flow
 import { Trans } from '@lingui/macro';
 
-import React, { Component } from 'react';
-import SelectField from '../../UI/SelectField';
+import * as React from 'react';
 import FlatButton from '../../UI/FlatButton';
-import SelectOption from '../../UI/SelectOption';
 import Dialog from '../../UI/Dialog';
-import { Column, Line } from '../../UI/Grid';
 import Window from '../../Utils/Window';
 import PreferencesContext from './PreferencesContext';
-import LocalesMetadata from '../../locales/LocalesMetadata';
 import { I18n } from '@lingui/react';
+import LanguageSelector from './LanguageSelector';
 
 type Props = {|
   open: boolean,
   onClose: (languageDidChange: boolean) => void,
 |};
 
-type State = {|
-  languageDidChange: boolean,
-|};
-
-const displayLocaleMetadata = localeMetadata => {
-  if (localeMetadata.languageCode === 'en') return false;
-  if (localeMetadata.languageCode === 'pseudo_LOCALE') return Window.isDev();
+const LanguageDialog = ({ open, onClose }: Props) => {
+  const { values } = React.useContext(PreferencesContext);
 
-  return true;
-};
+  const [languageDidChange, setLanguageDidChange] = React.useState<boolean>(
+    false
+  );
 
-const renderLanguageSelectOption = localeMetadata => {
-  const translationRatio = localeMetadata.translationRatio || 0;
-  const percent = (100 * localeMetadata.translationRatio).toFixed(0);
-  const isStarted = translationRatio > 0;
+  if (!open) return null;
 
   return (
-    <SelectOption
-      value={localeMetadata.languageCode}
-      primaryText={
-        localeMetadata.languageNativeName +
-        ' (' +
-        localeMetadata.languageName +
-        ')' +
-        (isStarted ? ` - ~${percent}%` : '')
-      }
-      disabled={!isStarted}
-      key={localeMetadata.languageCode}
-    />
+    <I18n>
+      {({ i18n }) => {
```

Comment:
random thought, I just realised we keep using i18n as a HoC.
Can we use a hook for translations? https://react.i18next.com/latest/usetranslation-hook
(maybe it's not in our version of i18n)

Reply:
We currently access i18n through https://github.com/4ian/lingui-react, I don't realize what is the complexity behind this.

Reply:
Oh I didn't realise we had our own version of lingui-react. not important for now, let's investigate later, this could be a nice refactoring over the codebase 👍

Reply:
Oh I feel you